### PR TITLE
wip

### PR DESCRIPTION
**Pull request by tkellen**
cc @phated do you think this revised functionality should just be in interpret?

### CODE REVIEW

**Comment by phated** [comment]
@tkellen I definitely think this could go in interpret.  it would allow for the tests to be co-located and it's just another method to expose, so interpret could continue being used as it is now but just expose the `prepare` method also.  thoughts?

**Comment by tkellen** [comment]
My only hesitation is that we'd be shipping interpret with liftoff but expecting consumers to provide their own. That seems weird.

**Comment by phated** [comment]
I was thinking we could make it the default but they could override if desired

**Comment by tkellen** [comment]
Hrm. I think I'm gonna leave them separate. If someone wants the loading behavior but also wants to provide an entirely different interpret object, I don't think they should have to inherit all of the existing ones.

**Comment by phated** [comment]
That's fair.  Want me to review/merge this?  Any idea why travis is failing?

**Comment by tkellen** [comment]
Yeah, a review/merge would be fantastic. Haven't looked into why this is failing but it's all green on my end.

**Comment by tkellen** [comment]
looks like i messed up the changelog, force pushing a fix for that in a sec.

**Comment by tkellen** [comment]
ready for review

**Comment by phated** [comment]
hmm, toml test is failing for me

**Comment by phated** [comment]
It looks like the requires aren't being cleaned up in the tests.  Let me dig in a bit

**Comment by tkellen** [comment]
bad variable scoping, perhaps? i just force pushed a fix. sorry, i should probably be doing fixups now.

**Comment by phated** [comment]
yeah, need to do fixups now.  It seems you removed the `afterEach` cleanup block

**Comment by tkellen** [comment]
Shouldn't be needed, but I could be wrong. Thanks for looking.

**Comment by phated** [comment]
When toml is throwing, iced-coffee-script and babel loaders are showing up in the stack trace.  not sure why things aren't being cleaned up.

**Comment by phated** [comment]
AHHA! Coffee-script seems to be hook `Module.prototype.load` in addition to `require.extensions`....ugh

**Comment by tkellen** [comment]
Shesh. Like every one of my modules for cli tools (interpret, v8flags, flagged-respawn, rechoir) cuts into the really gross underpinnings of node.

**Comment by phated** [comment]
Got almost everything sorted out but literate coffee/iced on the `.md` extension isn't working with my test alterations.  If I let coffee-script hijack the `load` stuff works but I want to get each test running from a clean cache.

**Comment by phated** [comment]
got it.  you are going to hate me :stuck_out_tongue:

**Comment by tkellen** [comment]
alright, alright, commit the code already. what abomination awaits?

File: index.js [hunk 1]
```diff
@@ -1,55 +1,47 @@
...
     cwd = path.dirname(path.resolve(filepath));
   }
-  var legacyModuleName = interpret.legacy[ext];
-  var config = interpret.configurations[moduleName];
-  var compiler;
-  if (legacyModuleName) {
-    compiler = handleLegacy(moduleName, legacyModuleName, cwd);
-  } else {
-    compiler = req(moduleName, cwd);
+  if (!Array.isArray(config)) {
```
Comment (phated) [inline index.js]: we can do this in the normalizer, 1 less isArray check

Comment (tkellen) [inline index.js]: That means the branch on line 17 will never be entered because it'll get `if(![undefined])`

Comment (phated) [inline index.js]: Ah, ok. I'll leave it

**Comment by phated** [comment]
@tkellen reassigning to you for review of all my changes. Your stuff looks good, just need a sanity check on all the test stuff.

**Comment by tkellen** [comment]
Looking good! Will cut a release in a min.

**Comment by tkellen** [comment]
PS: thanks!

**Pull request merged**